src/analysis/normed_space/finite_dimension.lean

```diff
@@ -174,6 +177,128 @@ def linear_equiv.to_continuous_linear_equiv [finite_dimensional 𝕜 E] (e : E 
   end,
   ..e }
 
+variables {ι : Type*} [fintype ι]
+
+/-- Construct a continuous linear map given the value at a finite basis. -/
+def is_basis.constrL {v : ι → E} (hv : is_basis 𝕜 v) (f : ι → F) :
+  E →L[𝕜] F :=
+⟨hv.constr f, begin
+  haveI : finite_dimensional 𝕜 E := finite_dimensional.of_finite_basis hv,
+  exact (hv.constr f).continuous_of_finite_dimensional,
+end⟩
+
+@[norm_cast] lemma is_basis.coe_constrL {v : ι → E} (hv : is_basis 𝕜 v) (f : ι → F) :
+  (hv.constrL f : E →ₗ[𝕜] F) = hv.constr f := rfl
+
+/-- The continuous linear equivalence between a vector space over `𝕜` with a finite basis and
+functions from its basis indexing type to `𝕜`. -/
+def is_basis.equiv_funL {v : ι → E} (hv : is_basis 𝕜 v) : E ≃L[𝕜] (ι → 𝕜) :=
+{ continuous_to_fun := begin
+    haveI : finite_dimensional 𝕜 E := finite_dimensional.of_finite_basis hv,
+    apply linear_map.continuous_of_finite_dimensional,
+  end,
+  continuous_inv_fun := begin
+    change continuous hv.equiv_fun.symm.to_fun,
+    apply linear_map.continuous_of_finite_dimensional,
+  end,
+  ..hv.equiv_fun }
+
+
+@[simp] lemma is_basis.constrL_apply {v : ι → E} (hv : is_basis 𝕜 v) (f : ι → F) (e : E) :
```

Comment:
I feel this lemma should be proved for the algebraic version (i.e., `hv.constr` without the `L`), probably with a `simp` attribute, and then the lemma for `hv.constrL` should follow (directly or by `simp`). Same thing for the next lemma.

Reply:
The situation is more complicated because `hv.constr` makes sense also in infinite dimension so everything is somehow duplicated, with lemmas about functions vs lemmas about finitely supported functions. But of course in this file I assume the source has finite dimension. So I added a finite dimensional version on the algebraic side.